plugin/storage/badger/spanstore/cache_test.go

```diff
@@ -107,10 +107,10 @@ func TestOldReads(t *testing.T) {
 
 // func runFactoryTest(tb testing.TB, test func(tb testing.TB, sw spanstore.Writer, sr spanstore.Reader)) {
 func runWithBadger(t *testing.T, test func(store *badger.DB, t *testing.T)) {
-	opts := badger.DefaultOptions
+	dir, _ := ioutil.TempDir("", "badger")
```

Comment:
With v2, we want to move away from the tempdirs and tmpfs. Instead, what should be used for in-memory situations is: ``badger.DefaultOptions("").WithInMemory(true)`` and this applies to tests also.

Reply:
@burmanm @jpkrohling In-memory mode would need more memory. Consider using tmpfs if you have memory constraints.

Reply:
tmpfs is what it currently uses - but to emulate the in-memory mode. Within docker containers I would personally opt for in-memory mode if such is available, given that some docker containers could be configured to map tmpfs to a swap or persistent storage, in which case the performance is a bit unpredictable. Or that they don't have tmpfs at all (that's possible also). 

Or does using in-memory mode prevent using compression or other methods that would reduce the memory usage?